types/validator_set.go

```diff
@@ -963,7 +961,7 @@ func ValidatorSetFromProto(vp *tmproto.ValidatorSet) (*ValidatorSet, error) {
 
 	vals.totalVotingPower = vp.GetTotalVotingPower()
 
-	return vals, nil
+	return vals, vals.ValidateBasic()
```

Comment:
Why validate here?

Reply:
assuming we've received it from untrusted source, we need to validate the input data

Reply:
Of course, but I'd argue that shouldn't happen during conversions - there are cases where we have incomplete structs that need to be converted, e.g. the problem with empty validator sets during test chain setup.